App/iOS/Delegates/SceneDelegate.swift

```diff
@@ -531,29 +541,52 @@ extension SceneDelegate: UIViewControllerRestoration {
 
 extension BrowserViewController {
   func handleReferralLookup(_ urp: UserReferralProgram) {
-
     if Preferences.URP.referralLookupOutstanding.value == true {
-      urp.referralLookup() { referralCode, offerUrl in
-        // Attempting to send ping after first urp lookup.
-        // This way we can grab the referral code if it exists, see issue #2586.
-        AppState.shared.dau.sendPingToServer()
-        if let code = referralCode {
-          let retryTime = AppConstants.buildChannel.isPublic ? 1.days : 10.minutes
-          let retryDeadline = Date() + retryTime
-
-          Preferences.NewTabPage.superReferrerThemeRetryDeadline.value = retryDeadline
-          
-          // TODO: Set the code in core somehow if we want to support Super Referrals again
-          //       then call updateSponsoredImageComponentIfNeeded
-        }
-
-        guard let url = offerUrl?.asURL else { return }
-        self.openReferralLink(url: url)
-      }
+      performProgramReferralLookup(urp, refCode: UserReferralProgram.getReferralCode())
     } else {
       urp.pingIfEnoughTimePassed()
     }
   }
+  
+  func handleSearchAdsInstallAttribution(_ urp: UserReferralProgram) {
+    urp.adCampaignLookup() { [weak self] response, error in
+      guard let self = self else { return }
+      
+      let refCode = self.generateReferralCode(attributionData: response, fetchError: error)
+      // Setting up referral code value
+      // This value should be set before first DAU ping
+      Preferences.URP.referralCode.value = refCode
+    }
+  }
+  
+  private func generateReferralCode(attributionData: AdAttributionData?, fetchError: Error?) -> String {
+    // Checking referral code from User Referral program exists If not send 001
+    // Prefix this code with BRV for organic iOS installs
+    var referralCode = "BRV\(UserReferralProgram.getReferralCode() ?? "001")"
```

Comment:
We do not use User referral program anymore but we can keep this code for now imo.
Can you change it so when its URP you do not append `BRV` to it? We did not append it in the past afaik.
So 3 cases in this method:
1. Normal install -> BRV001
2. Search Ads -> campaign code
3. User Referral program -> just whatever `UserReferralProgram.getReferralCode()` returns, no prefix

Reply:
Change at https://github.com/brave/brave-ios/pull/8194/commits/a78ff180fcc0bc02c35c0dd3f2451e3bdebeecf2 for removing